subsys/nrf_security/src/drivers/cracen/cracenpsa/src/cracen_mac_cmac.c

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <psa/crypto.h>
+#include <psa/crypto_values.h>
+#include <string.h>
+#include <sxsymcrypt/cmac.h>
+#include <sxsymcrypt/keyref.h>
+#include "common.h"
+#include <cracen/mem_helpers.h>
+#include "cracen_psa_primitives.h"
+
+#define AES_BLOCK_SIZE (16)
+
+psa_status_t cracen_cmac_setup(cracen_mac_operation_t *operation,
+				      const psa_key_attributes_t *attributes,
+				      const uint8_t *key_buffer, size_t key_buffer_size)
+{
+	int sx_status;
+	psa_status_t status = PSA_SUCCESS;
+
+	/* Only AES-CMAC is supported */
+	if (psa_get_key_type(attributes) != PSA_KEY_TYPE_AES) {
+		return PSA_ERROR_NOT_SUPPORTED;
+	}
+
+	if (key_buffer_size < AES_BLOCK_SIZE) {
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+
+	/* Load the key reference */
+	if (key_buffer_size > sizeof(operation->cmac.key_buffer)) {
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+
+	memcpy(operation->cmac.key_buffer, key_buffer, key_buffer_size);
+	status = cracen_load_keyref(attributes, operation->cmac.key_buffer, key_buffer_size,
+				    &operation->cmac.keyref);
+	if (status != PSA_SUCCESS) {
+		return status;
+	}
+
+	sx_status = sx_mac_create_aescmac(&operation->cmac.ctx, &operation->cmac.keyref);
+	if (sx_status) {
+		return silex_statuscodes_to_psa(sx_status);
+	}
```

Comment:
Should all of these be updated to `if (sx_status != SX_OK)` just so we are consistent through the driver?

Reply:
yeah, makes sense. I will update.